File: seed.py
"""装备数据字典种子脚本"""
from app.database import SessionLocal, engine, Base
from app.models import ItemDefinition, ItemCategory, ItemRarity

# 热门 CS2 饰品数据 (70+)
ITEM_DEFINITIONS = [
    # === 匕首 ===
    ("★ Karambit | Doppler", ItemCategory.knife, "Karambit", "Doppler", ItemRarity.covert, "★ Karambit | Doppler"),
    ("★ Karambit | Tiger Tooth", ItemCategory.knife, "Karambit", "Tiger Tooth", ItemRarity.covert, "★ Karambit | Tiger Tooth"),
    ("★ Karambit | Fade", ItemCategory.knife, "Karambit", "Fade", ItemRarity.covert, "★ Karambit | Fade"),
    ("★ Bayonet | Doppler", ItemCategory.knife, "Bayonet", "Doppler", ItemRarity.covert, "★ Bayonet | Doppler"),
    ("★ Bayonet | Rust Coat", ItemCategory.knife, "Bayonet", "Rust Coat", ItemRarity.covert, "★ Bayonet | Rust Coat"),
    ("★ M9 Bayonet | Doppler", ItemCategory.knife, "M9 Bayonet", "Doppler", ItemRarity.covert, "★ M9 Bayonet | Doppler"),
    ("★ M9 Bayonet | Fade", ItemCategory.knife, "M9 Bayonet", "Fade", ItemRarity.covert, "★ M9 Bayonet | Fade"),
    ("★ Butterfly Knife | Fade", ItemCategory.knife, "Butterfly Knife", "Fade", ItemRarity.covert, "★ Butterfly Knife | Fade"),
    ("★ Butterfly Knife | Doppler", ItemCategory.knife, "Butterfly Knife", "Doppler", ItemRarity.covert, "★ Butterfly Knife | Doppler"),
    ("★ Talon Knife | Doppler", ItemCategory.knife, "Talon Knife", "Doppler", ItemRarity.covert, "★ Talon Knife | Doppler"),
    ("★ Stiletto Knife | Doppler", ItemCategory.knife, "Stiletto Knife", "Doppler", ItemRarity.covert, "★ Stiletto Knife | Doppler"),
    ("★ Ursus Knife | Fade", ItemCategory.knife, "Ursus Knife", "Fade", ItemRarity.covert, "★ Ursus Knife | Fade"),
    ("★ Navaja Knife | Doppler", ItemCategory.knife, "Navaja Knife", "Doppler", ItemRarity.covert, "★ Navaja Knife | Doppler"),
    ("★ Classic Knife | Fade", ItemCategory.knife, "Classic Knife", "Fade", ItemRarity.covert, "★ Classic Knife | Fade"),

    # === AK-47 ===
    ("AK-47 | Redline", ItemCategory.weapon, "AK-47", "Redline", ItemRarity.classified, "AK-47 | Redline"),
    ("AK-47 | Vulcan", ItemCategory.weapon, "AK-47", "Vulcan", ItemRarity.covert, "AK-47 | Vulcan"),
    ("AK-47 | Fire Serpent", ItemCategory.weapon, "AK-47", "Fire Serpent", ItemRarity.covert, "AK-47 | Fire Serpent"),
    ("AK-47 | The Empress", ItemCategory.weapon, "AK-47", "The Empress", ItemRarity.covert, "AK-47 | The Empress"),
    ("AK-47 | Neon Rider", ItemCategory.weapon, "AK-47", "Neon Rider", ItemRarity.covert, "AK-47 | Neon Rider"),
    ("AK-47 | Bloodsport", ItemCategory.weapon, "AK-47", "Bloodsport", ItemRarity.covert, "AK-47 | Bloodsport"),
    ("AK-47 | Asiimov", ItemCategory.weapon, "AK-47", "Asiimov", ItemRarity.restricted, "AK-47 | Asiimov"),
    ("AK-47 | Ice Coaled", ItemCategory.weapon, "AK-47", "Ice Coaled", ItemRarity.restricted, "AK-47 | Ice Coaled"),
    ("AK-47 | Uncharted", ItemCategory.weapon, "AK-47", "Uncharted", ItemRarity.restricted, "AK-47 | Uncharted"),

    # === M4A4 ===
    ("M4A4 | Howl", ItemCategory.weapon, "M4A4", "Howl", ItemRarity.covert, "M4A4 | Howl"),
    ("M4A4 | Poseidon", ItemCategory.weapon, "M4A4", "Poseidon", ItemRarity.covert, "M4A4 | Poseidon"),
    ("M4A4 | Neo-Noir", ItemCategory.weapon, "M4A4", "Neo-Noir", ItemRarity.covert, "M4A4 | Neo-Noir"),
    ("M4A4 | The Emperor", ItemCategory.weapon, "M4A4", "The Emperor", ItemRarity.classified, "M4A4 | The Emperor"),
    ("M4A4 | Asiimov", ItemCategory.weapon, "M4A4", "Asiimov", ItemRarity.classified, "M4A4 | Asiimov"),
    ("M4A4 | Dragon King", ItemCategory.weapon, "M4A4", "Dragon King", ItemRarity.classified, "M4A4 | Dragon King"),

    # === M4A1-S ===
    ("M4A1-S | Printstream", ItemCategory.weapon, "M4A1-S", "Printstream", ItemRarity.covert, "M4A1-S | Printstream"),
    ("M4A1-S | Mecha Industries", ItemCategory.weapon, "M4A1-S", "Mecha Industries", ItemRarity.classified, "M4A1-S | Mecha Industries"),
    ("M4A1-S | Hyper Beast", ItemCategory.weapon, "M4A1-S", "Hyper Beast", ItemRarity.classified, "M4A1-S | Hyper Beast"),
    ("M4A1-S | Decimator", ItemCategory.weapon, "M4A1-S", "Decimator", ItemRarity.classified, "M4A1-S | Decimator"),
    ("M4A1-S | Golden Coil", ItemCategory.weapon, "M4A1-S", "Golden Coil", ItemRarity.classified, "M4A1-S | Golden Coil"),

    # === AWP ===
    ("AWP | Dragon Lore", ItemCategory.weapon, "AWP", "Dragon Lore", ItemRarity.covert, "AWP | Dragon Lore"),
    ("AWP | Gungnir", ItemCategory.weapon, "AWP", "Gungnir", ItemRarity.covert, "AWP | Gungnir"),
    ("AWP | Medusa", ItemCategory.weapon, "AWP", "Medusa", ItemRarity.covert, "AWP | Medusa"),
    ("AWP | Asiimov", ItemCategory.weapon, "AWP", "Asiimov", ItemRarity.covert, "AWP | Asiimov"),
    ("AWP | Oni Taiji", ItemCategory.weapon, "AWP", "Oni Taiji", ItemRarity.covert, "AWP | Oni Taiji"),
    ("AWP | Hyper Beast", ItemCategory.weapon, "AWP", "Hyper Beast", ItemRarity.covert, "AWP | Hyper Beast"),
    ("AWP | Fade", ItemCategory.weapon, "AWP", "Fade", ItemRarity.covert, "AWP | Fade"),
    ("AWP | Chrome Cannon", ItemCategory.weapon, "AWP", "Chrome Cannon", ItemRarity.classified, "AWP | Chrome Cannon"),

    # === Desert Eagle ===
    ("Desert Eagle | Blaze", ItemCategory.weapon, "Desert Eagle", "Blaze", ItemRarity.classified, "Desert Eagle | Blaze"),
    ("Desert Eagle | Code Red", ItemCategory.weapon, "Desert Eagle", "Code Red", ItemRarity.classified, "Desert Eagle | Code Red"),
    ("Desert Eagle | Printstream", ItemCategory.weapon, "Desert Eagle", "Printstream", ItemRarity.covert, "Desert Eagle | Printstream"),
    ("Desert Eagle | Ocean Drive", ItemCategory.weapon, "Desert Eagle", "Ocean Drive", ItemRarity.covert, "Desert Eagle | Ocean Drive"),
    ("Desert Eagle | Koi", ItemCategory.weapon, "Desert Eagle", "Koi", ItemRarity.classified, "Desert Eagle | Koi"),

    # === USP-S ===
    ("USP-S | Kill Confirmed", ItemCategory.weapon, "USP-S", "Kill Confirmed", ItemRarity.covert, "USP-S | Kill Confirmed"),
    ("USP-S | Printstream", ItemCategory.weapon, "USP-S", "Printstream", ItemRarity.covert, "USP-S | Printstream"),
    ("USP-S | The Traitor", ItemCategory.weapon, "USP-S", "The Traitor", ItemRarity.covert, "USP-S | The Traitor"),
    ("USP-S | Orion", ItemCategory.weapon, "USP-S", "Orion", ItemRarity.classified, "USP-S | Orion"),
    ("USP-S | Cortex", ItemCategory.weapon, "USP-S", "Cortex", ItemRarity.classified, "USP-S | Cortex"),

    # === Glock-18 ===
    ("Glock-18 | Fade", ItemCategory.weapon, "Glock-18", "Fade", ItemRarity.covert, "Glock-18 | Fade"),
    ("Glock-18 | Twilight Galaxy", ItemCategory.weapon, "Glock-18", "Twilight Galaxy", ItemRarity.classified, "Glock-18 | Twilight Galaxy"),
    ("Glock-18 | Bullet Queen", ItemCategory.weapon, "Glock-18", "Bullet Queen", ItemRarity.classified, "Glock-18 | Bullet Queen"),
    ("Glock-18 | Water Elemental", ItemCategory.weapon, "Glock-18", "Water Elemental", ItemRarity.classified, "Glock-18 | Water Elemental"),

    # === 手套 ===
    ("★ Sport Gloves | Pandora's Box", ItemCategory.gloves, "Sport Gloves", "Pandora's Box", ItemRarity.special, "★ Sport Gloves | Pandora's Box"),
    ("★ Sport Gloves | Vice", ItemCategory.gloves, "Sport Gloves", "Vice", ItemRarity.special, "★ Sport Gloves | Vice"),
    ("★ Driver Gloves | Crimson Weave", ItemCategory.gloves, "Driver Gloves", "Crimson Weave", ItemRarity.special, "★ Driver Gloves | Crimson Weave"),
    ("★ Hand Wraps | Cobalt Skulls", ItemCategory.gloves, "Hand Wraps", "Cobalt Skulls", ItemRarity.special, "★ Hand Wraps | Cobalt Skulls"),
    ("★ Moto Gloves | Spearmint", ItemCategory.gloves, "Moto Gloves", "Spearmint", ItemRarity.special, "★ Moto Gloves | Spearmint"),
    ("★ Moto Gloves | Boom!", ItemCategory.gloves, "Moto Gloves", "Boom!", ItemRarity.special, "★ Moto Gloves | Boom!"),
    ("★ Specialist Gloves | Fade", ItemCategory.gloves, "Specialist Gloves", "Fade", ItemRarity.special, "★ Specialist Gloves | Fade"),
    ("★ Bloodhound Gloves | Bronzed", ItemCategory.gloves, "Bloodhound Gloves", "Bronzed", ItemRarity.special, "★ Bloodhound Gloves | Bronzed"),

    # === MAC-10 ===
    ("MAC-10 | Neon Rider", ItemCategory.weapon, "MAC-10", "Neon Rider", ItemRarity.classified, "MAC-10 | Neon Rider"),
    ("MAC-10 | Stalker", ItemCategory.weapon, "MAC-10", "Stalker", ItemRarity.restricted, "MAC-10 | Stalker"),

    # === P250 ===
    ("P250 | Franklin", ItemCategory.weapon, "P250", "Franklin", ItemRarity.restricted, "P250 | Franklin"),

    # === Five-SeveN ===
    ("Five-SeveN | Fairy Tale", ItemCategory.weapon, "Five-SeveN", "Fairy Tale", ItemRarity.classified, "Five-SeveN | Fairy Tale"),

    # === AUG / SG 553 ===
    ("AUG | Akihabara Accept", ItemCategory.weapon, "AUG", "Akihabara Accept", ItemRarity.covert, "AUG | Akihabara Accept"),
    ("SG 553 | Integrale", ItemCategory.weapon, "SG 553", "Integrale", ItemRarity.covert, "SG 553 | Integrale"),

    # === FAMAS ===
    ("FAMAS | Eye of the Storm", ItemCategory.weapon, "FAMAS", "Eye of the Storm", ItemRarity.restricted, "FAMAS | Eye of the Storm"),

    # === 贴纸示例 ===
    ("Sticker | s1mple (Gold) | Antwerp 2022", ItemCategory.sticker, "Sticker", "s1mple (Gold)", ItemRarity.exceedingly_rare, "Sticker | s1mple (Gold) | Antwerp 2022"),
    ("Sticker | ZywOo (Gold) | Paris 2023", ItemCategory.sticker, "Sticker", "ZywOo (Gold)", ItemRarity.exceedingly_rare, "Sticker | ZywOo (Gold) | Paris 2023"),

    # === 探员 ===
    ("Agent | Ava (Imperial Officer)", ItemCategory.agent, "Agent", "Ava", ItemRarity.restricted, "Agent | Ava (Imperial Officer)"),
    ("Agent | The Elite Mr. Muhlik", ItemCategory.agent, "Agent", "The Elite Mr. Muhlik", ItemRarity.restricted, "Agent | The Elite Mr. Muhlik"),
]


def seed():
    """填充装备数据字典"""
    Base.metadata.create_all(bind=engine)
    db = SessionLocal()

    try:
        count = 0
        for name, category, weapon_type, skin_name, rarity, market_hash in ITEM_DEFINITIONS:
            exists = db.query(ItemDefinition).filter(
                ItemDefinition.market_hash_name == market_hash
            ).first()
            if not exists:
                db.add(ItemDefinition(
                    name=name,
                    category=category,
                    weapon_type=weapon_type,
                    skin_name=skin_name,
                    rarity=rarity,
                    market_hash_name=market_hash,
                ))
                count += 1

        db.commit()
        total = db.query(ItemDefinition).count()
        print(f"✅ Seed 完成: 新增 {count} 条, 总计 {total} 条装备定义")
    finally:
        db.close()


if __name__ == "__main__":
    print("🌱 正在填充装备数据字典...")
    seed()

File: prisma/seed.ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

const ITEM_DEFINITIONS = [
  // === 匕首 (Knives) ===
  { name: "★ Karambit | Doppler", category: "knife", weaponType: "Karambit", skinName: "Doppler", rarity: "covert", marketHashName: "★ Karambit | Doppler" },
  { name: "★ Karambit | Tiger Tooth", category: "knife", weaponType: "Karambit", skinName: "Tiger Tooth", rarity: "covert", marketHashName: "★ Karambit | Tiger Tooth" },
  { name: "★ Bayonet | Doppler", category: "knife", weaponType: "Bayonet", skinName: "Doppler", rarity: "covert", marketHashName: "★ Bayonet | Doppler" },
  { name: "★ Bayonet | Rust Coat", category: "knife", weaponType: "Bayonet", skinName: "Rust Coat", rarity: "covert", marketHashName: "★ Bayonet | Rust Coat" },
  { name: "★ M9 Bayonet | Doppler", category: "knife", weaponType: "M9 Bayonet", skinName: "Doppler", rarity: "covert", marketHashName: "★ M9 Bayonet | Doppler" },
  { name: "★ M9 Bayonet | Fade", category: "knife", weaponType: "M9 Bayonet", skinName: "Fade", rarity: "covert", marketHashName: "★ M9 Bayonet | Fade" },
  { name: "★ Butterfly Knife | Fade", category: "knife", weaponType: "Butterfly Knife", skinName: "Fade", rarity: "covert", marketHashName: "★ Butterfly Knife | Fade" },
  { name: "★ Butterfly Knife | Doppler", category: "knife", weaponType: "Butterfly Knife", skinName: "Doppler", rarity: "covert", marketHashName: "★ Butterfly Knife | Doppler" },
  { name: "★ Talon Knife | Doppler", category: "knife", weaponType: "Talon Knife", skinName: "Doppler", rarity: "covert", marketHashName: "★ Talon Knife | Doppler" },
  { name: "★ Stiletto Knife | Doppler", category: "knife", weaponType: "Stiletto Knife", skinName: "Doppler", rarity: "covert", marketHashName: "★ Stiletto Knife | Doppler" },
  { name: "★ Ursus Knife | Fade", category: "knife", weaponType: "Ursus Knife", skinName: "Fade", rarity: "covert", marketHashName: "★ Ursus Knife | Fade" },
  { name: "★ Navaja Knife | Doppler", category: "knife", weaponType: "Navaja Knife", skinName: "Doppler", rarity: "covert", marketHashName: "★ Navaja Knife | Doppler" },

  // === AK-47 ===
  { name: "AK-47 | Redline", category: "weapon", weaponType: "AK-47", skinName: "Redline", rarity: "classified", marketHashName: "AK-47 | Redline" },
  { name: "AK-47 | Vulcan", category: "weapon", weaponType: "AK-47", skinName: "Vulcan", rarity: "covert", marketHashName: "AK-47 | Vulcan" },
  { name: "AK-47 | Fire Serpent", category: "weapon", weaponType: "AK-47", skinName: "Fire Serpent", rarity: "covert", marketHashName: "AK-47 | Fire Serpent" },
  { name: "AK-47 | The Empress", category: "weapon", weaponType: "AK-47", skinName: "The Empress", rarity: "covert", marketHashName: "AK-47 | The Empress" },
  { name: "AK-47 | Neon Rider", category: "weapon", weaponType: "AK-47", skinName: "Neon Rider", rarity: "covert", marketHashName: "AK-47 | Neon Rider" },
  { name: "AK-47 | Bloodsport", category: "weapon", weaponType: "AK-47", skinName: "Bloodsport", rarity: "covert", marketHashName: "AK-47 | Bloodsport" },
  { name: "AK-47 | Asiimov", category: "weapon", weaponType: "AK-47", skinName: "Asiimov", rarity: "restricted", marketHashName: "AK-47 | Asiimov" },
  { name: "AK-47 | Ice Coaled", category: "weapon", weaponType: "AK-47", skinName: "Ice Coaled", rarity: "restricted", marketHashName: "AK-47 | Ice Coaled" },

  // === M4A4 ===
  { name: "M4A4 | Howl", category: "weapon", weaponType: "M4A4", skinName: "Howl", rarity: "covert", marketHashName: "M4A4 | Howl" },
  { name: "M4A4 | Poseidon", category: "weapon", weaponType: "M4A4", skinName: "Poseidon", rarity: "covert", marketHashName: "M4A4 | Poseidon" },
  { name: "M4A4 | Neo-Noir", category: "weapon", weaponType: "M4A4", skinName: "Neo-Noir", rarity: "covert", marketHashName: "M4A4 | Neo-Noir" },
  { name: "M4A4 | The Emperor", category: "weapon", weaponType: "M4A4", skinName: "The Emperor", rarity: "classified", marketHashName: "M4A4 | The Emperor" },
  { name: "M4A4 | Asiimov", category: "weapon", weaponType: "M4A4", skinName: "Asiimov", rarity: "classified", marketHashName: "M4A4 | Asiimov" },
  { name: "M4A4 | Dragon King", category: "weapon", weaponType: "M4A4", skinName: "Dragon King", rarity: "classified", marketHashName: "M4A4 | Dragon King" },

  // === M4A1-S ===
  { name: "M4A1-S | Knight", category: "weapon", weaponType: "M4A1-S", skinName: "Knight", rarity: "covert", marketHashName: "M4A1-S | Knight" },
  { name: "M4A1-S | Printstream", category: "weapon", weaponType: "M4A1-S", skinName: "Printstream", rarity: "covert", marketHashName: "M4A1-S | Printstream" },
  { name: "M4A1-S | Mecha Industries", category: "weapon", weaponType: "M4A1-S", skinName: "Mecha Industries", rarity: "classified", marketHashName: "M4A1-S | Mecha Industries" },
  { name: "M4A1-S | Hyper Beast", category: "weapon", weaponType: "M4A1-S", skinName: "Hyper Beast", rarity: "classified", marketHashName: "M4A1-S | Hyper Beast" },
  { name: "M4A1-S | Decimator", category: "weapon", weaponType: "M4A1-S", skinName: "Decimator", rarity: "classified", marketHashName: "M4A1-S | Decimator" },

  // === AWP ===
  { name: "AWP | Dragon Lore", category: "weapon", weaponType: "AWP", skinName: "Dragon Lore", rarity: "covert", marketHashName: "AWP | Dragon Lore" },
  { name: "AWP | Gungnir", category: "weapon", weaponType: "AWP", skinName: "Gungnir", rarity: "covert", marketHashName: "AWP | Gungnir" },
  { name: "AWP | Medusa", category: "weapon", weaponType: "AWP", skinName: "Medusa", rarity: "covert", marketHashName: "AWP | Medusa" },
  { name: "AWP | Asiimov", category: "weapon", weaponType: "AWP", skinName: "Asiimov", rarity: "covert", marketHashName: "AWP | Asiimov" },
  { name: "AWP | Oni Taiji", category: "weapon", weaponType: "AWP", skinName: "Oni Taiji", rarity: "covert", marketHashName: "AWP | Oni Taiji" },
  { name: "AWP | Hyper Beast", category: "weapon", weaponType: "AWP", skinName: "Hyper Beast", rarity: "covert", marketHashName: "AWP | Hyper Beast" },
  { name: "AWP | Fade", category: "weapon", weaponType: "AWP", skinName: "Fade", rarity: "covert", marketHashName: "AWP | Fade" },
  { name: "AWP | Chrome Cannon", category: "weapon", weaponType: "AWP", skinName: "Chrome Cannon", rarity: "classified", marketHashName: "AWP | Chrome Cannon" },

  // === Desert Eagle ===
  { name: "Desert Eagle | Blaze", category: "weapon", weaponType: "Desert Eagle", skinName: "Blaze", rarity: "classified", marketHashName: "Desert Eagle | Blaze" },
  { name: "Desert Eagle | Code Red", category: "weapon", weaponType: "Desert Eagle", skinName: "Code Red", rarity: "classified", marketHashName: "Desert Eagle | Code Red" },
  { name: "Desert Eagle | Printstream", category: "weapon", weaponType: "Desert Eagle", skinName: "Printstream", rarity: "covert", marketHashName: "Desert Eagle | Printstream" },
  { name: "Desert Eagle | Ocean Drive", category: "weapon", weaponType: "Desert Eagle", skinName: "Ocean Drive", rarity: "covert", marketHashName: "Desert Eagle | Ocean Drive" },
  { name: "Desert Eagle | Koi", category: "weapon", weaponType: "Desert Eagle", skinName: "Koi", rarity: "classified", marketHashName: "Desert Eagle | Koi" },

  // === USP-S ===
  { name: "USP-S | Kill Confirmed", category: "weapon", weaponType: "USP-S", skinName: "Kill Confirmed", rarity: "covert", marketHashName: "USP-S | Kill Confirmed" },
  { name: "USP-S | Printstream", category: "weapon", weaponType: "USP-S", skinName: "Printstream", rarity: "covert", marketHashName: "USP-S | Printstream" },
  { name: "USP-S | The Traitor", category: "weapon", weaponType: "USP-S", skinName: "The Traitor", rarity: "covert", marketHashName: "USP-S | The Traitor" },
  { name: "USP-S | Orion", category: "weapon", weaponType: "USP-S", skinName: "Orion", rarity: "classified", marketHashName: "USP-S | Orion" },
  { name: "USP-S | Cortex", category: "weapon", weaponType: "USP-S", skinName: "Cortex", rarity: "classified", marketHashName: "USP-S | Cortex" },

  // === Glock-18 ===
  { name: "Glock-18 | Fade", category: "weapon", weaponType: "Glock-18", skinName: "Fade", rarity: "covert", marketHashName: "Glock-18 | Fade" },
  { name: "Glock-18 | Twilight Galaxy", category: "weapon", weaponType: "Glock-18", skinName: "Twilight Galaxy", rarity: "classified", marketHashName: "Glock-18 | Twilight Galaxy" },
  { name: "Glock-18 | Bullet Queen", category: "weapon", weaponType: "Glock-18", skinName: "Bullet Queen", rarity: "classified", marketHashName: "Glock-18 | Bullet Queen" },
  { name: "Glock-18 | Water Elemental", category: "weapon", weaponType: "Glock-18", skinName: "Water Elemental", rarity: "classified", marketHashName: "Glock-18 | Water Elemental" },

  // === SSG 08 ===
  { name: "SSG 08 | Dragonfire", category: "weapon", weaponType: "SSG 08", skinName: "Dragonfire", rarity: "classified", marketHashName: "SSG 08 | Dragonfire" },
  { name: "SSG 08 | Death Strike", category: "weapon", weaponType: "SSG 08", skinName: "Death Strike", rarity: "mil-spec", marketHashName: "SSG 08 | Death Strike" },

  // === 手套 (Gloves) ===
  { name: "★ Sport Gloves | Pandora's Box", category: "gloves", weaponType: "Sport Gloves", skinName: "Pandora's Box", rarity: "special", marketHashName: "★ Sport Gloves | Pandora's Box" },
  { name: "★ Sport Gloves | Vice", category: "gloves", weaponType: "Sport Gloves", skinName: "Vice", rarity: "special", marketHashName: "★ Sport Gloves | Vice" },
  { name: "★ Driver Gloves | Crimson Weave", category: "gloves", weaponType: "Driver Gloves", skinName: "Crimson Weave", rarity: "special", marketHashName: "★ Driver Gloves | Crimson Weave" },
  { name: "★ Hand Wraps | Cobalt Skulls", category: "gloves", weaponType: "Hand Wraps", skinName: "Cobalt Skulls", rarity: "special", marketHashName: "★ Hand Wraps | Cobalt Skulls" },
  { name: "★ Moto Gloves | Spearmint", category: "gloves", weaponType: "Moto Gloves", skinName: "Spearmint", rarity: "special", marketHashName: "★ Moto Gloves | Spearmint" },
  { name: "★ Moto Gloves | Boom!", category: "gloves", weaponType: "Moto Gloves", skinName: "Boom!", rarity: "special", marketHashName: "★ Moto Gloves | Boom!" },
  { name: "★ Specialist Gloves | Fade", category: "gloves", weaponType: "Specialist Gloves", skinName: "Fade", rarity: "special", marketHashName: "★ Specialist Gloves | Fade" },

  // === MAC-10 ===
  { name: "MAC-10 | Neon Rider", category: "weapon", weaponType: "MAC-10", skinName: "Neon Rider", rarity: "classified", marketHashName: "MAC-10 | Neon Rider" },
  { name: "MAC-10 | Stalker", category: "weapon", weaponType: "MAC-10", skinName: "Stalker", rarity: "restricted", marketHashName: "MAC-10 | Stalker" },

  // === P250 ===
  { name: "P250 | Franklin", category: "weapon", weaponType: "P250", skinName: "Franklin", rarity: "restricted", marketHashName: "P250 | Franklin" },

  // === Five-SeveN ===
  { name: "Five-SeveN | Fairy Tale", category: "weapon", weaponType: "Five-SeveN", skinName: "Fairy Tale", rarity: "classified", marketHashName: "Five-SeveN | Fairy Tale" },
];

async function main() {
  console.log("🌱 Seeding item definitions...");

  for (const def of ITEM_DEFINITIONS) {
    await prisma.itemDefinition.upsert({
      where: { marketHashName: def.marketHashName },
      update: {},
      create: def,
    });
  }

  const count = await prisma.itemDefinition.count();
  console.log(`✅ Seeded ${count} item definitions`);
}

main()
  .catch((e) => {
    console.error("❌ Seed failed:", e);
    process.exit(1);
  })
  .finally(() => prisma.$disconnect());
